### DISCUSSION TITLE
Have I hit a dead end?

### DISCUSSION THREAD

**Discussion by sam-wheat** [unoplatform/uno#11385] [Q&A]
I've run into the errors below when I run [my project](https://github.com/leaderanalytics/LeaderPivot.XAML.Uno) under linux.  It seems to run OK under Windows although it is far from complete.

Working WPF implementation of the project I am trying to port to Uno can be found [here](https://github.com/leaderanalytics/LeaderPivot.XAML.WPF).  WPF implementation uses a library for some effects which I have removed from the Uno version.  I don't think I am using any animations or anything beyond basic controls in the Uno implementation.  

Other than reading [documentation that appears relevent](https://platform.uno/docs/articles/wpf-migration.html) and searching for keyworlds I have not spent a lot of time troubleshooting this.  Partly because I don't know where to begin and partly because I suspect I have run into a dead end.  Should I reasonably expect to be able to port this project to Uno?  

> fail: Microsoft.UI.Xaml.UIElement[0]
>       The member Vector3 UIElement.Translation is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.UIElement.Vector3%20UIElement.Translation
> fail: Microsoft.UI.Xaml.Controls.Primitives.ComboBoxTemplateSettings[0]
>       The member ComboBoxTemplateSettings.DropDownContentMinWidth is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.Controls.Primitives.ComboBoxTemplateSettings.ComboBoxTemplateSettings.DropDownContentMinWidth
> fail: Uno.UI.DataBinding.BindingPropertyHelper[0]
>       The [ScrollViewer] property getter does not exist on type [Microsoft.UI.Xaml.Controls.ComboBox]
> fail: Microsoft.UI.Xaml.BrushTransition[0]
>       The member BrushTransition.BrushTransition() is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.BrushTransition.BrushTransition.BrushTransition%28%29
> fail: Microsoft.UI.Xaml.BrushTransition[0]
> 		The member TimeSpan BrushTransition.Duration is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.BrushTransition.TimeSpan%20BrushTransition.Duration
> fail: Microsoft.UI.Xaml.Controls.ContentPresenter[0]
>       The member BrushTransition ContentPresenter.BackgroundTransition is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.Controls.ContentPresenter.BrushTransition%20ContentPresenter.BackgroundTransition
> fail: Microsoft.UI.Xaml.Media.Animation.SplitCloseThemeAnimation[0]
>       The member SplitCloseThemeAnimation.SplitCloseThemeAnimation() is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.Media.Animation.SplitCloseThemeAnimation.SplitCloseThemeAnimation.SplitCloseThemeAnimation%28%29
> fail: Microsoft.UI.Xaml.Controls.Primitives.ComboBoxTemplateSettings[0]
>       The member ComboBoxTemplateSettings.DropDownOffset is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.Controls.Primitives.ComboBoxTemplateSettings.ComboBoxTemplateSettings.DropDownOffset
> fail: Microsoft.UI.Xaml.Controls.Primitives.ComboBoxTemplateSettings[0]
>       The member ComboBoxTemplateSettings.DropDownOpenedHeight is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.Controls.Primitives.ComboBoxTemplateSettings.ComboBoxTemplateSettings.DropDownOpenedHeight
> fail: Microsoft.UI.Xaml.Media.Animation.SplitCloseThemeAnimation[0]
>       The member void Begin() is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.Media.Animation.SplitCloseThemeAnimation.void%20Begin%28%29
> fail: Microsoft.UI.Xaml.Controls.ScrollContentPresenter[0]
>       The member float ZoomFactor is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=Microsoft.UI.Xaml.Controls.ScrollContentPresenter.float%20ZoomFactor

**Comment by jeromelaban** [comment]
Thanks for the question. Some of the pieces are not yet implemented, but may not be a problem. Can you provide a sample of what is not working for you?

**Reply by sam-wheat** [reply]
Hi @jeromelaban,

When I run the project under Windows it renders as expected.  Under linux I basically get a blank screen.

Windows:
![image](https://user-images.githubusercontent.com/16885210/220117643-d30aa523-345b-46f8-9915-dba8ef1a2497.png)

Linux:
![image](https://user-images.githubusercontent.com/16885210/220117998-cd91380e-80e0-4c0c-b17e-79d1a0023308.png)

**Reply by jeromelaban** [reply]
I see. This may look like a measuring issue, since the outer border is rendering. You can try specifying Width/Height properties on some of your controls to see where the issue may come from.

**Reply by leaderanalytics** [reply]
Thank you @jeromelaban, for a control like this is is not really possible to constrain the size of any container except the outer one.  I tried putting height/width on the control itself but it has no effect.

**Reply by jeromelaban** [reply]
Thanks. I understand that it's not a solution for your app, but it would help to determine where the issue is. The best is to try to narrow down what is causing the UI to measure incorrectly, by adding explicit dimensions on some controls to see where the issue could be.
